src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -6919,6 +6920,209 @@ def create_stack(self, stack: StackRequest) -> StackResponse:
+                            else:
+                                resource_id = None
+                                resource_type = requirements.resource_type
+                                if requirements.resource_id_attr is not None:
+                                    resource_id = (
+                                        component_info.configuration.get(
+                                            requirements.resource_id_attr
+                                        )
+                                    )
+
+                            satisfied, msg = requirements.is_satisfied_by(
+                                connector=service_connector,
+                                component=component,
+                            )
+
+                            if not satisfied:
+                                raise ValueError(
+                                    "Please pick a connector that is "
+                                    "compatible with the component flavor and "
+                                    "try again.."
+                                )
+
+                            if not resource_id:
+                                if service_connector.resource_id:
+                                    resource_id = service_connector.resource_id
+                                elif service_connector.supports_instances:
+                                    raise ValueError(
+                                        f"Multiple {resource_type} resources "
+                                        "are available for the selected "
+                                        "connector. Please use a `resource_id` "
+                                        "to configure a "
+                                        f"{resource_type} resource."
+                                    )
+
+                            component_update = ComponentUpdate(
+                                connector=service_connector.id,
+                                connector_resource_id=resource_id,
+                            )
+                            self.update_stack_component(
+                                component_id=component.id,
+                                component_update=component_update,
+                            )
+
+                    components_mapping[component_type] = [
+                        component.id,
+                    ]
+
+                # Stack
+                stack_name = full_stack.name
+                while True:
+                    try:
+                        stack_request = StackRequest(
+                            user=full_stack.user,
+                            workspace=full_stack.workspace,
+                            name=stack_name,
+                            description=full_stack.description,
+                            components=components_mapping,
+                        )
+                        stack_response = self.create_stack(stack_request)
```

Comment:
This now means the `REGISTERED_STACK` gets sent twice, right?

Reply:
Yes. Creating the full stack can potentially mean that you have to create service connectors, components, and a stack. To avoid code duplication, I used the methods that were already provided by the `SqlZenStore`.

However, as you wrote, this leads to a place where we have two `REGISTERED_STACK` calls happening within the boundaries of the same function. Potentially, we can solve this in different ways:

1. We can change the event to a different one, which is specific to this endpoint. Perhaps something like `REGISTERED_FULL_STACK`?
2. Or, we can copy over the code of the `create_stack` into the `create_full_stack` method so that we don't call the same analytics event twice.

Reply:
Or maybe you don't call the `REGISTERED_STACK` in your full stack method if it doesn't create a stack. This is a helper method, and the actual stack is being registered in `create_stack(...)`, which correctly calls the analytics event.

Reply:
Why not just add a `metadata` field to the stack request and DB schema, where the `generated_by_wizard` can be stored? This would be the right approach IMO, that solves both this problem and also others like detecting which stacks were created by the wizard later on.

Reply:
@schustmi As we discussed before, we have solved this over the `labels` field (previously present in components and connectors, now available for stacks). We also introduced a new notion of labels which start with `zenml:` which directly gets added to the analytics metadata. Would be great if you could take a quick look once again.

Reply:
LGTM